upi/aws/cloudformation/02_cluster_infra.yaml

```diff
@@ -253,6 +270,80 @@ Resources:
       Runtime: "python3.7"
       Timeout: 120
 
+  RegisterSubnetTagsLambdaIamRole:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: !Join ["-", [!Ref ClusterName, "subnet-tags-lambda-role"]]
```

Comment:
nit: Better to use `InfrastructureName` for this and other resources, since that allows us to wash any troublesome-for-AWS characters, length, or non-uniqueness from the cluster name.  But doesn't have to happen in this PR.